File: forloop-add-test/forloop_add_test.pl
#!/usr/bin/env perl

use strict;

my $toval = int($ARGV[0]);
my $sumval = 0;

for (my $i=0; $i < $toval; $i++) {
    for (my $j=0; $j < $toval; $j++) {
        $sumval += $i+($j^1);
    }
}

print 'Perl: ', $sumval, "\n";

File: forloop-add-test/forloop_add_test.rb
#!/usr/bin/env ruby

def main(toval)
    sumval = 0
    for i in 0 .. toval-1
        for j in 0 .. toval-1
            sumval += i+(j^1)
        end
    end
    print 'Ruby: ', sumval, "\n"
end

main ( ARGV[0].to_i )
